src/mesh/Draco_Mesh.hh

```diff
@@ -0,0 +1,126 @@
+ *    b) and a boundary layout with side and off-process linkage.
+ * 2) Geometry, which implies a metric for distance between points.
+ *
+ * Possibly temporary features:
+ * 1) The cell_type_ vector (argument to the constructor) is currently taken to
+ *    be the number of nodes per cell.
+ * 2) The layout data structure(s) will proabably be moved to a separate class,
+ *    where accessors might be used on a flattened version.
+ */
+//===========================================================================//
+
+class Draco_Mesh {
+public:
+  // >>> TYPEDEFS
+  typedef rtt_mesh_element::Geometry Geometry;
+  typedef std::map<unsigned,
+                   std::vector<std::pair<unsigned, std::vector<unsigned>>>>
+      Layout;
+  typedef std::map<unsigned, std::vector<std::pair<int, std::vector<unsigned>>>>
+      Boundary_Layout;
+
+private:
+  // >>> DATA
+
+  // Dimension
+  const unsigned dimension;
+
+  // Geometry enumeration
+  const Geometry geometry;
+
+  // Number of cells
+  const unsigned num_cells;
+
+  // Number of nodes
+  const unsigned num_nodes;
+
+  // Side set flag (can be used for mapping BCs to sides)
+  const std::vector<unsigned> side_set_flag;
+
+  // Layout of mesh: vector index is cell index, vector element is
+  // description of cell's adjacency to other cells in the mesh.
+  Layout cell_to_cell_linkage;
+  // TODO: update this to a full layout class.
+
+  // Boundary layout of mesh
+  Boundary_Layout cell_to_side_linkage;
+
+  // vector subscripted with node index with coordinate vector
+  std::vector<std::vector<double>> node_coord_vec;
+
+public:
+  //! Constructor.
+  Draco_Mesh(unsigned dimension_, Geometry geometry_,
+             const std::vector<unsigned> &cell_type_,
+             const std::vector<unsigned> &cell_to_node_linkage_,
+             const std::vector<unsigned> &side_set_flag_,
+             const std::vector<unsigned> &side_node_count_,
+             const std::vector<unsigned> &side_to_node_linkage_,
+             const std::vector<double> &coordinates_,
+             const std::vector<unsigned> &global_node_number_);
+
+  // >>> ACCESSORS
+
+  unsigned get_dimension() const { return dimension; }
+  Geometry get_geometry() const { return geometry; }
+  unsigned get_num_cells() const { return num_cells; }
+  unsigned get_num_nodes() const { return num_nodes; }
+  const Layout &get_cc_linkage() const { return cell_to_cell_linkage; }
+  const Boundary_Layout &get_cs_linkage() const { return cell_to_side_linkage; }
+
+  // >>> SERVICES
+
+private:
+  // >>> SUPPORT FUNCTIONS
+
+  //! Calculate the cell-to-cell linkage (layout)
+  // TODO: add layout class and complete temporary version of this function.
+  void compute_cell_to_cell_linkage(
+      const std::vector<unsigned> &cell_type,
+      const std::vector<unsigned> &cell_to_node_linkage,
+      const std::vector<unsigned> &side_node_count,
+      const std::vector<unsigned> &side_to_node_linkage);
+
```

Comment:
One function (that computes the vector of node coordinates) was changed to const and moved to the initializer list.  The other unfortunately calculates two data members; unless these are changed to one data member (consisting of a `std::tuple`) it does not seem to be possible.  As you pointed out, this might prematurely obfuscate the layout structures.  A note about this is in the most recent comment message.